src/methods/analyzing/addRank.ts

```diff
@@ -0,0 +1,68 @@
+import { SimpleDataItem } from "../../types/SimpleData.types.js"
+import sortValues from "./sortValues.js"
+
+export default function addRank(
+    data: SimpleDataItem[],
+    rankTitle: string,
+    rankBy?: string | string[],
+    sort?: true | false,
+    locale?: string | (string | undefined | null | boolean)[],
+    handleTies?: "sequential" | "indexWithTie" | "indexWithoutTie"
+): SimpleDataItem[] {
+    if (!rankBy) {
+        data.map((datum, i) => {
+            datum[rankTitle] = i + 1
+        })
+    } else {
+        const sortData = sort ? data : data.slice(0)
```

Comment:
I think you can work directly with the incoming data. The user would expect the output to be sorted.

Reply:
sortInPlace implemented to adress this